tensorflow/python/eager/def_function_test.py

```diff
@@ -240,6 +241,28 @@ def apply(self, x):
     m1 = MyModel()
     self.assertAllEqual(m1.apply(3.0), 6.0)
 
+  @unittest.expectedFailure
+  def testMethodAllowDynamicVariableWithoutGuards(self):
+
+    class Foo:
+
+      def __init__(self):
+        self._var = 0
+
+      def __call__(self, val):
+        self.compute(val)
+        return self._var
+
+      @def_function.function
+      def compute(self,val):
+        self._var = variables.Variable(val)
+       
+    def_function.ALLOW_DYNAMIC_VARIABLE_CREATION = True
+    foo = Foo()
+    self.assertAllEqual(foo(0.3), 0.3)
+    self.assertAllEqual(foo(0.9), 0.9,
+        "https://github.com/tensorflow/tensorflow/issues/27120")
```

Comment:
@mihaimaruseac @mdanatg What do you think might be the best approach when we have a 1-to-N relation between a ticket and its failing tests. 
I don't want that this test will fail at some point as this case will be solved but other expecting failing test connected to the same issue are still passing.
For the nature of these tests do you think that that a team member will need to just grep this URL and looking for other related expecting failing tests in the source code before evaluating to close the mentioned bug?

It might work but it still seems a bit too manual and someone might forget to do it.

Reply:
I think the risk of having stale tests that still fail even after the bug is marked fixed is low. Most of the time, the PR which fixes the bug should cause the expectedFailure to raise an error. If we end up seeing a lot, we can think of a more automated way to identify them.

Reply:
I am more worried about silently passing expected failing tests connected to the same issue. 
E.g. Something like an internal or contributor PR that don't cover all the testing use cases and so It cannot close the related github issue and instead It will close It.

The alternative is to have more atomic github issues for tracking the use cases that we are testing more strictly with a 1-to-1 approach.

Reply:
I meant it is really related on our atomicity concept of a PR as a transatcion of all the expected failing tests connected to a single issue.